app/src/main/java/com/eveningoutpost/dexdrip/Services/DexShareCollectionService.java

```diff
@@ -289,6 +289,11 @@ public void call(EGVRecord[] egvRecords) {
                                         if (egvRecords != null) {
                                             Log.d(TAG, "Made the full round trip, got " + egvRecords.length + " EVG Records");
                                             BgReading.create(egvRecords, addativeSystemTimeOffset, getApplicationContext());
+                                            {
```

Comment:
wouldn't a logic of:
lock
try {

} finaly
{
free lock
}

be more clear, and with a smaller chance to break?

On Tue, Jul 21, 2015 at 9:47 PM, Stephen Black <notifications@github.com>
wrote:

> In
> app/src/main/java/com/eveningoutpost/dexdrip/Services/DexShareCollectionService.java
> <https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/104#discussion_r35138858>
> :
>
> > @@ -289,6 +289,11 @@ public void call(EGVRecord[] egvRecords) {
> >                                          if (egvRecords != null) {
> >                                              Log.d(TAG, "Made the full round trip, got " + egvRecords.length + " EVG Records");
> >                                              BgReading.create(egvRecords, addativeSystemTimeOffset, getApplicationContext());
> > +                                            {
>
> is this block doing anything?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/StephenBlackWasAlreadyTaken/xDrip-Experimental/pull/104/files#r35138858>
> .
>